proto/accessgraph/v1alpha/graph.proto

```diff
@@ -48,4 +50,6 @@ message Edge {
   string to = 2;
   // edge_type is the type of the edge, e.g. "member_of", "belongs_to", etc.
   string edge_type = 3;
+  // properties is dynamically typed set of edge properties, with the schema varying from one edge type to another.
+  google.protobuf.Struct properties = 4;
```

Comment:
Without having a type how UI can read that field?

Reply:
Note that there is no schema for the json representation of edge property in the database either. You do have information about the edge type, however, and that gives you the expected schema implicitly.

The alternatives are either plain string, with embedded JSON or other encoding, or going the other way with strongly typed proto messages.

I can easily do either 🤷 , but I think that Struct is the sweet spot for efficiency, compact representation and flexibility. I didn't think we want to change the protos each time a slight change is made in the edge properties.

Reply:
As we talked today. If you keep the different set separate inside the payload here UI will be able to deserialize them.